core/src/client/async_client/mod.rs

```diff
@@ -489,6 +491,11 @@ async fn background_task<S: TransportSenderT, R: TransportReceiverT>(
 						let _ = stop_subscription(&mut sender, &mut manager, unsub).await;
 					}
 				}
+				// Subscription error response.
+				else if let Ok(response) = serde_json::from_str::<SubscriptionError<_>>(&raw) {
+					tracing::debug!("[backend]: recv subscription closed {:?}", response);
+					let _ = process_subscription_close_response(&mut manager, response);
```

Comment:
So here, the backend has sent back something like this I think?:

```
{ jsonrpc: "2.0", params: { subscription: "abc123", error: { code, message, data } } }
```

As a result, we close the subscription on the client. Right now, we don't pass on that error to the client though? (this may have been what you were getting at in a previous comment and I'm just catching up now :))

Reply:
yes, because I was lazy because o the lifetime issues with serde.

Reply:
OK that makes sense; if it's a bit of a pain to do, let's add an issue to address this separately if it's not important to Substrate!

Reply:
no this is just the client code so shouldn't matter